model/task_history.go

```diff
@@ -351,16 +352,12 @@ func (iter *taskHistoryIterator) GetChunk(v *Version, numBefore, numAfter int, i
 	return chunk, nil
 }
 
-func (self *taskHistoryIterator) GetDistinctTestNames(numCommits int) ([]string, error) {
-	session, mdb, err := db.GetGlobalSessionFactory().GetSession()
-	if err != nil {
-		return nil, errors.Wrap(err, "problem getting database session")
-	}
-	defer session.Close()
-
-	session.SetSocketTimeout(time.Minute)
-
-	pipeline := mdb.C(task.Collection).Pipe(
+func (self *taskHistoryIterator) GetDistinctTestNames(env evergreen.Environment, numCommits int) ([]string, error) {
+	ctx, cancel := env.Context()
+	defer cancel()
+	opts := options.Aggregate().SetMaxTime(time.Minute)
+	fmt.Println("Using new changes")
+	pipeline, err := env.DB().Collection(task.Collection).Aggregate(ctx,
```

Comment:
could you name this variable something else? It returns a cursor, not a pipeline